tests/unit/Type/NonEmptyVecTypeTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Tests\Unit\Type;
+
+use Psl\Collection;
+use Psl\Dict;
+use Psl\Iter;
+use Psl\Str;
+use Psl\Type;
+use Psl\Vec;
+
+/**
+ * @extends TypeTest<non-empty-list<mixed>>
+ */
+final class NonEmptyVecTypeTest extends TypeTest
+{
+    public function getType(): Type\TypeInterface
+    {
+        return Type\non_empty_vec(Type\int());
+    }
+
+    public function getValidCoercions(): iterable
+    {
+        yield [
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        ];
+
+        yield [
+            ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'],
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        ];
+
+        yield [
+            new Collection\Vector([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        ];
+
+        yield [
+            new Collection\Map([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        ];
+
+        yield [
+            new Collection\Vector(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10']),
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        ];
+
+        yield [
+            new Collection\Map(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10']),
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        ];
+
+        yield [
+            Dict\map_keys(Vec\range(1, 10), static fn(int $key): string => (string)$key),
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        ];
+
+        yield [
+            Dict\map(Vec\range(1, 10), static fn(int $value): string => Str\format('00%d', $value)),
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        ];
+
+        yield [
+            ['a' => 1, 'b' => 2, 'c' => 3, 'd' => 4, 'e' => 5],
+            [1, 2, 3, 4, 5]
+        ];
+    }
+
+    public function getInvalidCoercions(): iterable
+    {
+        yield [[]];
+        yield [1.0];
+        yield [1.23];
+        yield [Type\bool()];
+        yield [null];
+        yield [false];
+        yield [true];
+        yield [STDIN];
+    }
+
+    public function getToStringExamples(): iterable
+    {
+        yield [$this->getType(), 'non-empty-list<int>'];
```

Comment:
should be non-empty-vec 🤔 while psalm calls this type 'list', we call it 'vec' here, since list cannot be an actual type in PHP due to list being a reserved keyword.

Reply:
`VecType` uses `list<>` instead of `vec<>`, does that need updating too?

Reply:
yes